lib/compare.ex

```diff
@@ -0,0 +1,25 @@
+defprotocol Compare do
```

Comment:
We can implement a protocol only once per type, correct? That might be a problem when trying to have closed intervals with different order in the same VM.

Reply:
That is actually something I use: I have a project with different `ord` implementations on the same type.